spec/active_record/connection_adapters/abstract/immutable_statements_spec.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+describe ActiveRecord::ConnectionAdapters::ImmutableStatements do
+  subject { PersonMigration.new }
+
+  describe '#add_immutable_column' do
+    after do
+      subject.add_immutable_column(
+        table_name,
+        column_options: column_options,
+        foreign_key_options: foreign_key_options,
+        index_options: index_options
+      )
+    end
+
+    let(:table_name) { 'people' }
+    let(:column_name) { ActiveRecord::Immutable::DEFAULT_PREVIOUS_VERSION_ID_COLUMN_NAME }
+    let(:column_options) { { column_test: 'column_test' } }
+    let(:foreign_key_options) { { foreign_key_test: 'foreign_key_test', column: column_name } }
+    let(:index_options) { { index_test: 'index_test' } }
+    let(:migration_instance) { ActiveRecord::ConnectionAdapters::SchemaStatements }
+
+    it 'should create a column, foreign key and index' do
```

Comment:
What do you think about changing the `should create` to `creates` to follow the Better Specs guideline? http://www.betterspecs.org/#should

Reply:
Also, can we break these expectations into three separate tests? In case of failure, it would be easier to identify which behavior failed.